src/index.ts

```diff
@@ -0,0 +1,52 @@
+import { resolve, join } from 'path';
+import { readJson } from 'fs-extra';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import remoteTags from 'remote-git-tags';
+
+const REQUIRED_TAG_FORMAT = '$version';
+
+// eslint-disable-next-line camelcase
+const cloneUrl: string = github.context.payload.repository?.clone_url;
+const tagFormat: string = core.getInput('tag-format');
+
+async function getVersion(): Promise<string> {
+  const packagePath = join(core.getInput('package-directory'), 'package.json');
+  const { version } = await readJson(resolve(packagePath));
+
+  if (!version) {
+    throw new Error(`Missing "version" in ${packagePath}`);
+  }
+
+  return version;
+}
+
+async function run(): Promise<void> {
+  if (!tagFormat.includes(REQUIRED_TAG_FORMAT)) {
+    throw new Error(
+      `tag-format is missing require "${REQUIRED_TAG_FORMAT}" pattern`,
+    );
+  }
+
+  const version = await getVersion();
+  const tags = await remoteTags(cloneUrl);
+  const tagName = tagFormat.replace('$version', version);
+
+  if (core.isDebug()) {
+    console.log('Available remote tags:\n', tags.keys());
+  }
+
+  if (tags.has(tagName)) {
+    throw new Error(`Tag "${tagName} already exists in ${cloneUrl}.`);
+  }
+
+  core.info(`Tag "${tagName} is available to use.`);
+}
+
+function handleError(err: any): void {
```

Comment:
Any better way to specify the type of this error other than `any`?

Reply:
`Error` is a type representing an Error object created through `new Error()`

Reply:
So I should change that to `Error`? I was seeing somewhere that I shouldn't use primitive types. Maybe this doesn't count for Error? Is this fine if the error is an extension of Error?

Reply:
Yeah it believe should be fine to just use Error, that is what I have done in other TS code bases. I think the "not use primitive types" bit is more about String and Object, but not totally sure.

Reply:
Okay, I'll try that out and report back if I see something different.